bokeh/models/annotations.py

```diff
@@ -107,6 +108,89 @@ class Legend(Annotation):
         Dict(String, List(Instance(GlyphRenderer))), lambda d: list(d.items())
     )
 
+class Arrow(Annotation):
+    """ Render an arrow as an annotation.
+
+    """
+
+    tail_x = NumberSpec(help="""
+    The x-coordinates to locate the arrow tails.
+    """)
+
+    tail_y = NumberSpec(help="""
+    The y-coordinates to locate the arrow tails.
+    """)
+
+    head_x = NumberSpec(help="""
+    The x-coordinates to locate the arrow heads.
+    """)
+
+    head_y = NumberSpec(help="""
+    The y-coordinates to locate the arrow heads.
+    """)
+
+    head_size = Float(default=25, help="""
+    The size, in pixels, of the arrow head.
+    """)
+
+    tail_units = Enum(SpatialUnits, default='data', help="""
+    The unit type for the head_x and head_y attributes. Interpreted as "data
+    space" units by default.
```

Comment:
I wonder if there are any use cases where a user might want to specify units for `x` and `y` independently? Also the grouping of props here is a bit weird I would put all the "head" props and all the "tail" props together.